includes/class-wc-moneybadger-gateway.php

```diff
@@ -0,0 +1,429 @@
+            array( $this, 'check_moneybadger_webhook' )
+        );
+        add_action(
+            'woocommerce_api_wc_gateway_moneybadger_order_status',
+            array( $this, 'check_if_order_is_paid' )
+        );
+    }
+
+    /**
+     * Initialize the admin fields for the plugin settings.
+     */
+    public function init_form_fields() {
+        $this->form_fields = array(
+            'enabled' => array(
+                'title'   => __('Enable/Disable', 'wc-gateway-moneybadger'),
+                'type'    => 'checkbox',
+                'label'   => __('Enable the MoneyBadger Payment Gateway', 'wc-gateway-moneybadger'),
+                'default' => 'yes',
+            ),
+            'title' => array(
+                'title'       => __('Title', 'wc-gateway-moneybadger'),
+                'type'        => 'text',
+                'description' => __(
+                    'The payment method title displayed to customers at checkout.',
+                    'wc-gateway-moneybadger')
+                ,
+                'default'     => __('MoneyBadger', 'wc-gateway-moneybadger'),
+                'desc_tip'    => true,
+            ),
+            'description' => array(
+                'title'       => __('Description', 'wc-gateway-moneybadger'),
+                'type'        => 'textarea',
+                'description' => __(
+                    'The payment method description displayed to customers at checkout.',
+                    'wc-gateway-moneybadger'
+                ),
+                'default'     => __('Pay securely with Bitcoin Lightning, VALR, Luno and Binance wallets via MoneyBadger.', 'wc-gateway-moneybadger'),
+            ),
+            'merchant_code' => array(
+                'title'       => __('Merchant Code', 'wc-gateway-moneybadger'),
+                'type'        => 'text',
+                'description' => __('Enter your MoneyBadger merchant code.', 'wc-gateway-moneybadger'),
+                'default'     => '',
+            ),
+            'api_key' => array(
+                'title'       => __('API Key', 'wc-gateway-moneybadger'),
+                'type'        => 'password',
+                'description' => __('Enter your MoneyBadger API key.', 'wc-gateway-moneybadger'),
+                'default'     => '',
+            ),
+            'testmode' => array(
+                'title'       => __('Test mode', 'wc-gateway-moneybadger'),
+                'type'        => 'checkbox',
+                'label'       => __('Enable test mode', 'wc-gateway-moneybadger'),
+                'default'     => 'yes',
+            ),
+        );
+    }
+
+    /**
+     * Generate the MoneyBadger payment iframe URL.
+     *
+     * @param string $order_id  The order ID
+     * @return string
+     */
+    public function get_moneybadger_iframe_url($order_id) {
+        $amount = $this->get_order_total();
+        $reference = $order_id . '-' . time();
```

Comment:
I'd prefer to use a random string for this rather than something predictable

` bin2hex(random_bytes(16));` or something similar